### PR TITLE
Improve location parsing

### PR DESCRIPTION
**Pull request by maniac103**
Move it to the correct place, and make it more lenient to also cover the 'latitude,longitude,elevation' case.

Fixes issue reported in https://community.openhab.org/t/mapview-in-android-app-does-not-show-the-location/73379

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/model/ParsedState.java [hunk 1]
```diff
@@ -212,6 +218,28 @@ private static NumberState parseAsNumber(String state, String format) {
...
+    private static Location parseAsLocation(String state) {
+        String[] splitState = state.split(",");
+        // Valid states are either "latitude,longitude" or "latitude,longitude,elevation",
+        // (we ignore elevation in the latter case)
+        if (splitState.length == 2 || splitState.length == 3) {
+            try {
+                Location l = new Location("openhab");
+                l.setLatitude(Float.valueOf(splitState[0]));
+                l.setLongitude(Float.valueOf(splitState[1]));
+                l.setTime(System.currentTimeMillis());
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/model/ParsedState.java]: The time when the location was changed in openHAB isn't the time when the user starts the app. I suggest to not set the time here.

**Comment by mueller-ma** [comment]
Can you build and upload a beta apk?

File: mobile/src/main/java/org/openhab/habdroid/model/ParsedState.java (hunk 1 above)
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/model/ParsedState.java]: I basically only set the time to make the debug output look nicer, as `Location.toString()` prints some question marks if it isn't set ;-)

**Comment by mueller-ma** [comment]
The binding that causes this issue might always set the third parameter to "0", but other bindings might use it. I prefer to pass it to the MapView.

**Comment by maniac103** [comment]
> I prefer to pass it to the MapView

How do you want to do that? MapView only accepts LatLng instance, which are latitude/longitude pairs ;-)

**Comment by mueller-ma** [comment]
https://developer.android.com/reference/android/location/Location.html#setAltitude(double)
I haven't looked at the Maps API.

**Comment by maniac103** [comment]
Oops, missed that. I'll set it; but maps API indeed doesn't use it.

File: mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.java [hunk 2]
```diff
@@ -186,17 +187,11 @@ private static void setMarker(GoogleMap map, LatLng position, Item item,
...
-                try {
-                    return new LatLng(Float.valueOf(splitState[0]), Float.valueOf(splitState[1]));
-                } catch (NumberFormatException e) {
-                    // ignored
-                }
-            }
-            return null;
+        private static LatLng toLatLng(ParsedState state) {
+            Location location = state != null ? state.asLocation() : null;
+            return state != null
```
Comment (mueller-ma) [inline mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.java]: Need to check for location == null here.

Comment (maniac103) [reply mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.java]: Oops. It's `!=` actually, but thanks, will fix this.

**Comment by mueller-ma** [comment]
Do we want to add a common (full and foss) no-position handling? Like hiding the map and display an "empty view", like in notification list?

**Comment by maniac103** [comment]
> Do we want to add a common (full and foss) no-position handling? Like hiding the map and display an "empty view", like in notification list?

For foss I surely don't ... what's the point in showing a 'no map support' view for every MapView? For full we could do that, but I think that's outside of scope of this PR.

**Pull request merged**